### PR TITLE
add task solution

### PR DESCRIPTION
**Pull request by Kozubowicz**
[DEMO LINK](https://Kozubowicz.github.io/react_list-of-goods/)

### CODE REVIEW

File: src/App.tsx [hunk 1]
```diff
@@ -15,81 +15,113 @@ export const goodsFromServer = [
...
+    listOfGoods = [...goodsFromServer];
+  } else if (state.sortType === 'ALPHABET') {
+    listOfGoods = [...goodsFromServer].sort();
+  } else if (state.sortType === 'LENGTH') {
+    listOfGoods = [...goodsFromServer]
+      .sort((el, el2) => (el.length - el2.length));
+  }
+
+  if (state.isReversed) {
+    listOfGoods = [...listOfGoods].reverse();
```
Comment (volodymyr-soltys97) [inline src/App.tsx:86]:
```suggestion
    listOfGoods.reverse();
```

File: src/App.tsx [hunk 2]
```diff
@@ -15,81 +15,113 @@ export const goodsFromServer = [
...
+  }
 
-export const App: React.FC = () => {
   return (
     <div className="section content">
       <div className="buttons">
         <button
           type="button"
-          className="button is-info is-light"
+          className={`button is-info ${state.sortType === 'ALPHABET' ? '' : 'is-light'}`}
```
Comment (volodymyr-soltys97) [inline src/App.tsx:94]: Use a classnames library for add classes with condition

File: src/App.tsx [hunk 3]
```diff
@@ -15,81 +15,113 @@ export const goodsFromServer = [
...
+          className={`button is-success ${state.sortType === 'LENGTH' ? '' : 'is-light'}`}
+          onClick={sortByLength}
         >
           Sort by length
         </button>
 
         <button
           type="button"
-          className="button is-warning is-light"
+          className={`button is-warning ${state.isReversed === true ? '' : 'is-light'}`}
```
Comment (volodymyr-soltys97) [inline src/App.tsx:110]:
```suggestion
          className={`button is-warning ${state.isReversed ? '' : 'is-light'}`}
```

File: src/App.tsx [hunk 4]
```diff
@@ -15,81 +15,113 @@ export const goodsFromServer = [
...
-export function getReorderedGoods(
-  goods: string[],
-  { sortType, isReversed }: ReorderOptions,
-) {
-  // To avoid the original array mutation
-  const visibleGoods = [...goods];
+  const sortAlphabetically = () => {
+    setState((prevState) => ({
+      ...prevState,
+      sortType: 'ALPHABET',
```
Comment (volodymyr-soltys97) [inline src/App.tsx:29]: Consider will be better if you create a variable for all magic words and use it everywhere

**Review by volodymyr-soltys97 — CHANGES_REQUESTED** [review]
Good job 👍 
Let's improve your code

**Review by nazarmatsevych — APPROVED** [review]
Good work!

**Pull request closed without merging**